scipy/sparse/tests/test_base.py

```diff
@@ -4300,7 +4300,10 @@ def spmatrix(self, D, **kwargs):
                                            zero_pos[0][k],
                                            zero_pos[1][k])
 
-        arg1 = self._arg1_for_noncanonical(M)
+        if sorted_indices:
+            arg1 = self._arg1_for_noncanonical(M, sorted_indices)
+        else:
+            arg1 = self._arg1_for_noncanonical(M)
```

Comment:
I see. I think the cleanest solution is to make all subclasses accept the `sorted_indices` kwarg, even if they don't use it. Also I don't see a reason for the current duplication in `_arg1_for_noncanonical` between `_NonCanonicalCompressedMixin` and `_NonCanonicalCSMixin`. BSR should be tested with both sorted and unsorted indices, too.